src/Games/NexusMods.Games.AdvancedInstaller.UI/AdvancedManualInstallerUI.cs

```diff
@@ -17,6 +17,13 @@ namespace NexusMods.Games.AdvancedInstaller.UI;
 // ReSharper disable once InconsistentNaming
 public class AdvancedManualInstallerUI : IAdvancedInstallerHandler
 {
+    /// <summary>
+    /// If true, the UI is not running and the installer will quietly fail.
+    ///
+    /// NOTE(halgari): this is an ugly hack, but we don't have a better tool for it at the moment.
+    /// </summary>
+    public static bool Headless { get; set; } = false;
```

Comment:
If this ever becomes a larger issue, then I think one possible pattern here is using DI.

We could use DI to inject an interface with 'debug' (or whatever you wanna call them) flags.

One of these flags could be `IsUIAvailable`, so if we're running unit tests, that could be used to enable headless mode as to not risk getting tests stuck.

This is just one possible approach that comes to mind.

Reply:
Yep, we can and should look into this.